matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/network/Request.kt

```diff
@@ -21,36 +21,51 @@ import kotlinx.coroutines.delay
 import org.matrix.android.sdk.api.failure.Failure
 import org.matrix.android.sdk.api.failure.shouldBeRetried
 import org.matrix.android.sdk.internal.network.ssl.CertUtil
-import retrofit2.Call
-import retrofit2.awaitResponse
+import retrofit2.HttpException
 import timber.log.Timber
 import java.io.IOException
 
+// To use when there is no init block to provide
 internal suspend inline fun <DATA : Any> executeRequest(globalErrorReceiver: GlobalErrorReceiver?,
-                                                        block: Request<DATA>.() -> Unit) = Request<DATA>(globalErrorReceiver).apply(block).execute()
+                                                        noinline requestBlock: suspend () -> DATA): DATA {
+    return executeRequest(globalErrorReceiver, requestBlock, {})
+}
+
+internal suspend inline fun <DATA : Any> executeRequest(globalErrorReceiver: GlobalErrorReceiver?,
+                                                        noinline requestBlock: suspend () -> DATA,
+                                                        initBlock: Request<DATA>.() -> Unit): DATA {
+    return Request(globalErrorReceiver, requestBlock)
+            .apply(initBlock)
+            .execute()
+}
 
-internal class Request<DATA : Any>(private val globalErrorReceiver: GlobalErrorReceiver?) {
+internal class Request<DATA : Any>(
```

Comment:
Don't think there's much need for this class anymore. Might as well merge it into the `executeRequest` function.
The private `var`s would become local variables and the public `var`s would become function parameters.

Reply:
Yes, I agree, it's a bit dirty like that

Reply:
Done, thanks